src/inmanta/server/services/resourceservice.py

```diff
@@ -412,9 +413,12 @@ async def mark_deployed(
             send_events=False,
             keep_increment_cache=True,
             is_increment_notification=True,
+            connection=connection,
         )
 
-    async def get_increment(self, env: data.Environment, version: int) -> tuple[abc.Set[ResourceIdStr], abc.Set[ResourceIdStr]]:
+    async def get_increment(
+        self, env: data.Environment, version: int, connection: Optional[Connection] = None
```

Comment:
This is a tricky one: we move it into a transaction but we keep a cache that lives outside it. I have to give this more thought.

Reply:
A cache we can hopefully get rid off soon and that
1.  we should I believe always miss (and populate) (we do the release, so we are first)
2. I would hope the transaction will give us exclusivity on our own code path for our own environment


So 
1. we can never read it, and we can't get blocked on it. 
2. We should never have to abort the transaction, so we can't make it stale
3. Only after the transaction commits, will the version become available for pull, so only then the cache line will be read. (any stale read will invalidate the cache)

Reply:
I don't completely follow this: we invalidate the cache but also write it, persisting the transaction state before it commits (if it does).

Reply:
if we roll back, 
 - the version number we released will never be pulled, so the cache is invalid for every intent and purpose
 - the release will happen again later, re-populating the cache

Reply:
Ok, I see, thanks.